tests/components/stream/test_worker.py

```diff
@@ -363,6 +368,9 @@ async def test_skip_missing_dts(hass):
     bad_packet_start = int(LONGER_TEST_SEQUENCE_LENGTH / 2)
     num_bad_packets = MAX_MISSING_DTS - 1
     for i in range(bad_packet_start, bad_packet_start + num_bad_packets):
+        if packets[i].is_keyframe:
+            num_bad_packets -= 1
```

Comment:
This variable could use a different name now. My impression is the # of bad packets is the same, but now the ratio of packets to output segments is a little different.  Maybe it should be expressed in terms of # of expected output video segment rather than the number of skipped packets to subtract from the total?

Reply:
I'm surprised.more tests didn't need to be changed when changing the keyfrsmes.

Reply:
The # of "bad packets" is reduced since we are continuing before setting the dts to None

Reply:
I guess it doesn't affect much since it only deals with a little bit of variability in the segment length (the difference between the min segment length and the next keyframe), and most of that is already taken into account by adjusting the `SEGMENT_DURATION`.

Reply:
I looked into it a bit and made a tweak that corrects the `SEGMENT_DURATION` calculation in the worker and adds the `KEYFRAME_RATE` parameter

Reply:
I think i may have been confused, thinking this was counting segments.  (I was thinking before segments and packets were 1:1). I see this really just applies to video packets.